x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -173,14 +173,18 @@ export type RuleNameOverrideOrUndefined = t.TypeOf<typeof ruleNameOverrideOrUnde
 export const status = t.keyof({ open: null, closed: null, 'in-progress': null });
 export type Status = t.TypeOf<typeof status>;
 
-export const job_status = t.keyof({
-  succeeded: null,
-  failed: null,
-  'going to run': null,
-  'partial failure': null,
-  warning: null,
-});
-export type JobStatus = t.TypeOf<typeof job_status>;
+export enum RuleExecutionStatus {
+  'succeeded' = 'succeeded',
+  'failed' = 'failed',
+  'going to run' = 'going to run',
+  'partial failure' = 'partial failure',
+  /**
+   * @deprecated 'partial failure' status should be used instead
+   */
+  'warning' = 'warning',
+}
+
+export const ruleExecutionStatus = enumeration('RuleExecutionStatus', RuleExecutionStatus);
```

Comment:
`'partial failure' = 'partial failure',` is needed for compatibility with the data structure we receive with API responses. Otherwise, we had to write data converters `API string -> RuleExecutionStatus`.